airbyte-integrations/connectors/destination-bigquery-denormalized/src/test/java/io/airbyte/integrations/destination/bigquery/formatter/DefaultBigQueryDenormalizedRecordFormatterTest.java

```diff
@@ -0,0 +1,359 @@
+import static io.airbyte.integrations.destination.bigquery.util.BigQueryDenormalizedTestSchemaUtils.getSchemaWithReferenceDefinition;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.Field.Mode;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import io.airbyte.integrations.base.JavaBaseConstants;
+import io.airbyte.integrations.destination.bigquery.BigQuerySQLNameTransformer;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+
+class DefaultBigQueryDenormalizedRecordFormatterTest {
+
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  private static Stream<Arguments> actualAndExpectedSchemasProvider() {
+    return Stream.of(
+        arguments(getSchema(), getExpectedSchema()),
+        arguments(getSchemaWithFormats(), getExpectedSchemaWithFormats()),
+        arguments(getSchemaWithDateTime(), getExpectedSchemaWithDateTime()),
+        arguments(getSchemaWithInvalidArrayType(), getExpectedSchemaWithInvalidArrayType()),
+        arguments(getSchemaWithNestedDatetimeInsideNullObject(),
+            getExpectedSchemaWithNestedDatetimeInsideNullObject()),
+        arguments(getSchemaArrays(), getExpectedSchemaArrays()));
+  }
+
+  @ParameterizedTest
+  @MethodSource("actualAndExpectedSchemasProvider")
+  void testDefaultSchema(final JsonNode schemaToProcess, final JsonNode expectedSchema) {
+    DefaultBigQueryDenormalizedRecordFormatter rf = new DefaultBigQueryDenormalizedRecordFormatter(
+        schemaToProcess, new BigQuerySQLNameTransformer());
+
+    assertEquals(expectedSchema, rf.formatJsonSchema(schemaToProcess));
+  }
+
+  @Test
+  void testSchema() {
+    final JsonNode jsonNodeSchema = getSchema();
+    DefaultBigQueryDenormalizedRecordFormatter rf = new DefaultBigQueryDenormalizedRecordFormatter(
+        jsonNodeSchema, new BigQuerySQLNameTransformer());
+
+    final Schema bigQuerySchema = rf.getBigQuerySchema(jsonNodeSchema);
+
+    final FieldList fields = bigQuerySchema.getFields();
```

Comment:
is it possible to construct an `expectedSchema` object and just `assertEquals(expectedSchema, bigQuerySchema)`? This approach looks pretty verbose and easy to introduce errors (or accidentally forget to assert something)

Reply:
how about just updating the first and last test methods in each class? I think that would help future code authors make good choices when they add new tests, without requiring a rewrite on every test method.

Reply:
I made a refactoring for Default and GCS formate's tests. 
- expected schemas are used now 
- now`GcsBigQueryDenormalizedRecordFormatterTest` cover only it's own functionality